dbms/src/Common/BackgroundTask.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/BackgroundTask.h>
+
+#include <fstream>
+namespace DB
+{
+bool process_mem_usage(double & vm_usage, double & resident_set)
+{
+    vm_usage = 0.0;
+    resident_set = 0.0;
+
+    // 'file' stat seems to give the most reliable results
+    std::ifstream stat_stream("/proc/self/stat", std::ios_base::in);
```

Comment:
Does it work for non-linux OS?

Reply:
`isProcStatSupported` will be called before to check whether it is support "/proc/self/stat", if not, RSS track and check will be disabled.